vignettes/glue.Rmd

```diff
@@ -1,140 +1,206 @@
+glue("Release was on a {format(release_date, '%A')}.")
 ```
 
-All valid R code works in expressions, including braces and
-escaping.Backslashes do need to be doubled just like in all R strings.
-Consider, for instance, adding a variable named `` foo}` `` and printing
-its value:
-
-```{r}
-`foo}\`` <- "My value"
-glue("{
-  `foo}\\``
-     }")
+## Control of line breaks
 
-```
+`glue()` honors the line breaks in its input:
 
 ```{r}
-`foo}\`` <- "My value"
-glue("{
-      {
-        '}\\'' # { and } in comments, single quotes
-        \"}\\\"\" # or double quotes are ignored
-        `foo}\\`` # as are { in backticks
-      }
-  }")
+glue("
+  A formatted string
+  Can have multiple lines
+    with additional indention preserved
+  "
+)
 ```
 
-## Long lines of text
+The example above demonstrates some other important facts about the pre-processing of the template string:
+
+-   An empty first or last line is automatically trimmed.
+-   Leading whitespace that is common across all lines is trimmed.
 
-Glue comes with a number of features that make it easier to use when
-work with large quantities of text. Leading whitespace and blank lines
-from the first and last lines are automatically trimmed with `glue()`,
-letting you indent the strings naturally in code:
+The elimination of common leading whitespace is advantageous, because you aren't forced to choose between indenting your code normally and getting the output you actually want.
+This is easier to appreciate when you have `glue()` inside a function body (this example also shows an alternative way of styling the end of a `glue()` call):
 
 ```{r}
-my_fun <- function() {
+foo <- function() {
   glue("
     A formatted string
     Can have multiple lines
-      with additional indention preserved
-    ")  
+      with additional indention preserved")
 }
-my_fun()
+foo()
 ```
 
-An add extra newlines can be used if you want a leading or trailing
-newline:
+On the other hand, what if you don't want a line break in the output, but you also like to limit the length of lines in your source code to, e.g., 80 characters?
+The first option is to use `\\` to break the template string into multiple lines, without getting line breaks in the output:
 
 ```{r}
+release_date <- as.Date("2017-06-13")
 glue("
+  The first version of the glue package was released on \\
+  a {format(release_date, '%A')}.")
+```
+
+This comes up fairly often when an expression to evaluate inside `{}` takes up more characters than its result, i.e. `format(release_date, '%A')` versus `Tuesday`.
+A second way to achieve the same result is to break the template into individual pieces, which are then concatenated.
+
+```{r}
+glue(
+  "The first version of the glue package was released on ",
+  "a {format(release_date, '%A')}."
+)
+```
+
+If you want an explicit newline at the start or end, include an extra empty line.
+
+```{r}
+# no leading or trailing newline
+x <- glue("
+  blah
+  ")
+unclass(x)
```

Comment:
Did you mention `unclass()` already?

Reply:
No. I'm going it here just to see the `\n` explicitly. If you just print the result, the added newline at the front is obvious, but the one at the back is not because printing gives you a newline effectively regardless.

Reply:
Are you saying that I should address the whole issue of glue's class and printing?

Reply:
I went with this interpretation.